### PR TITLE
Product db

### PR DESCRIPTION
**Pull request by okrdima**
Add product with DB,fix more bugs, rewrite measures

### CODE REVIEW

**Review by olenashevchuk — APPROVED** [review]

File: src/app/domains/Cart/routes/CartEdit/CartEdit.layout.js [hunk 1]
```diff
@@ -1,41 +1,64 @@
...
+        category: data.category,
+        price: data.price,
+        quantity: data.quantity,
+        measures: data?.measures || '',
+        dateBuy: data.dateBuy
+      })
+      history.goBack()
+    } catch (error) {
+      console.log(error)
+    }
```
Comment (olenashevchuk) [inline src/app/domains/Cart/routes/CartEdit/CartEdit.layout.js:51]: I think you need to use the Message here

File: src/app/domains/Measure/components/combined/MeasureModalWithForm/MeasureModalWithForm.template.js [hunk 2]
```diff
@@ -1,22 +1,56 @@
...
+    try {
+      const { measureSelect } = data
+      setLoading(true)
+      await deleteData(COLLECTIONS.MEASURES, measureSelect.id)
+      form.reset({})
+    } catch (error) {
+      console.log(error)
+    }
+    setLoading(false)
+    setSwitchState(true)
```
Comment (olenashevchuk) [inline src/app/domains/Measure/components/combined/MeasureModalWithForm/MeasureModalWithForm.template.js:33]: add the Message

File: src/app/domains/Measure/components/combined/MeasureModalWithForm/MeasureModalWithForm.template.js [hunk 3]
```diff
@@ -1,22 +1,56 @@
...
+      setLoading(true)
+      const id = firestore.collection(COLLECTIONS.MEASURES).doc().id
+      await setData(COLLECTIONS.MEASURES, id, {
+        id: id,
+        measure: data.measure
+      })
+      form.reset()
+    } catch (error) {
+      console.log(error)
+    }
```
Comment (olenashevchuk) [inline src/app/domains/Measure/components/combined/MeasureModalWithForm/MeasureModalWithForm.template.js:47]: and here

File: src/app/domains/RegularProduct/components/table/RegularProductsTable/RegularProductsTable.template.js [hunk 4]
```diff
@@ -1,60 +1,26 @@
...
-    const recievedData = value?.docs.map((item) => {
-      return { id: item.id, ...item.data() }
-    })
-    setProducts(recievedData)
-    return () => {}
-  }, [value])
-
-  return <>{products && <Table type="regular" products={products} />}</>
+  return (
+    <>
```
Comment (GingerBoy14) [inline src/app/domains/RegularProduct/components/table/RegularProductsTable/RegularProductsTable.template.js:14]: add Spinner on loading

File: src/app/domains/RegularProduct/routes/RegularProductEdit/RegularProductEdit.layout.js [hunk 5]
```diff
@@ -1,18 +1,75 @@
...
+    } catch (error) {
+      console.log(error)
+    }
+  }
+  if (loading) {
+    return <Spinner />
+  }
+  if (!dataForDefaultValue) {
+    return <Spinner />
+  }
```
Comment (GingerBoy14) [inline src/app/domains/RegularProduct/routes/RegularProductEdit/RegularProductEdit.layout.js:55]: Simplify this to 1 condition

File: src/app/domains/RegularProduct/routes/RegularProductShow/RegularProductShow.layout.js [hunk 6]
```diff
@@ -26,16 +34,30 @@ const RegularProductShow = (props) => {
...
-      type="product"
-      {...props}
-      id={id}
-      data={product}
-      dropdownItem={qr}
-    />
+  return loading ? (
+    <Spinner />
+  ) : (
+    <>
```
Comment (GingerBoy14) [inline src/app/domains/RegularProduct/routes/RegularProductShow/RegularProductShow.layout.js:41]: Remove this condition

File: src/app/domains/Wish/components/table/WishTable.template.js [hunk 7]
```diff
@@ -4,10 +4,22 @@ import { firestore } from 'app/services/Firestore'
...
 
 const WishTable = (props) => {
+  const { setStatusMessage } = props
+
   const { actions } = props
   const [data] = useCollectionData(firestore.collection(COLLECTIONS.WISHES))
+
   return (
-    <>{data && <Table type="wishes" products={data} actions={actions} />}</>
+    <>
```
Comment (GingerBoy14) [inline src/app/domains/Wish/components/table/WishTable.template.js:13]: Add loading, use single destructorize

File: src/app/domains/Wish/routes/WishEdit/WishEdit.layout.js [hunk 8]
```diff
@@ -1,39 +1,65 @@
...
+      console.log(error)
+    }
   }
 
-  const onSubmitButton = () => history.goBack()
   const onCancel = () => history.goBack()
 
+  if (!dataForDefaultValue) {
+    return <Spinner />
+  }
```
Comment (GingerBoy14) [inline src/app/domains/Wish/routes/WishEdit/WishEdit.layout.js:59]: use one condition for loading

File: src/app/domains/Wish/routes/WishShow/WishShow.layout.js [hunk 9]
```diff
@@ -1,20 +1,51 @@
...
+  })
+  const handleClose = () => {
+    setStatusMessage({ open: false, message: '', type: '' })
   }
-  return <ProductAdvancedView type="wish" data={value.data()} id={id} />
+
+  return loading ? (
+    <Spinner />
+  ) : (
+    <>
```
Comment (GingerBoy14) [inline src/app/domains/Wish/routes/WishShow/WishShow.layout.js:30]: Replace to separate condition

**Review by GingerBoy14 — CHANGES_REQUESTED** [review]

File: src/app/domains/Cart/components/table/CartTable.template.js [hunk 10]
```diff
@@ -1,17 +1,24 @@
...
-  const data = value?.docs.map((item) => ({
-    id: item.id,
-    ...item.data()
-  }))
+  const [data] = useCollectionData(firestore.collection(COLLECTIONS.CART))
 
-  return <>{value && <Table type="cart" products={data} />}</>
+  return (
+    <>
+      {data && (
```
Comment (GingerBoy14) [inline src/app/domains/Cart/components/table/CartTable.template.js:13]: add loading

File: src/app/domains/Cart/routes/CartEdit/CartEdit.layout.js [hunk 11]
```diff
@@ -1,41 +1,64 @@
...
   }
 
-  const onSubmitButton = () => history.goBack()
+  if (!dataForDefaultValue) {
+    return <Spinner />
+  }
   const onCancel = () => history.goBack()
```
Comment (GingerBoy14) [inline src/app/domains/Cart/routes/CartEdit/CartEdit.layout.js:57]: Move loading to one condition

File: src/app/domains/Cart/routes/CartShow/CartShow.layout.js [hunk 12]
```diff
@@ -1,21 +1,51 @@
...
+  })
+  const handleClose = () => {
+    setStatusMessage({ open: false, message: '', type: '' })
   }
-  return <ProductAdvancedView type="cart" data={value.data()} id={id} />
+
+  return loading ? (
+    <Spinner />
+  ) : (
+    <>
```
Comment (GingerBoy14) [inline src/app/domains/Cart/routes/CartShow/CartShow.layout.js:30]: move loading to separate condition

File: src/app/domains/Purchase/routes/PurchaseAll/PurchaseAll.layout.js [hunk 13]
```diff
@@ -1,6 +1,6 @@
-import { PurchaseList } from '../../components/list'
+import { PurchaseTable } from '../../components/table'
```
Comment (GingerBoy14) [inline src/app/domains/Purchase/routes/PurchaseAll/PurchaseAll.layout.js:1]: fix import

**Review by GingerBoy14 — CHANGES_REQUESTED** [review]

**Review by GingerBoy14 — APPROVED** [review]

**Pull request merged**